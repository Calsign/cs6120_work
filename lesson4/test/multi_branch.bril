@main() {
  x: int = const 10;
  z: int = const 20;
  cond: bool = const true;
  br cond .br1 .br2;

.br1:
  x: int = const 11;
  jmp .mid;

.br2:
  z: int = const 21;
  x: int = const 11;
  br cond .mid .br4;

.mid:
  z: int = const 22;
  br cond .br3 .br4;

.br3:
  y: int = id z;
  x: int = const 12;
  jmp .end;

.br4:
  z: int = id y;
  jmp .end;

.end:
  print x y z;
}
