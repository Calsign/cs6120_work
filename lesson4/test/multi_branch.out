{"op":"const","value":10,"dest":"x","type":"int"}  ->  x: 10
{"op":"const","value":20,"dest":"z","type":"int"}  ->  x: 10, z: 20
{"op":"const","value":true,"dest":"cond","type":"bool"}  ->  cond: true, x: 10, z: 20
{"op":"br","args":["cond"],"labels":["br1","br2"]}  ->  cond: true, x: 10, z: 20
{"op":"const","value":11,"dest":"x","type":"int"}  ->  cond: true, x: 11, z: 20
{"op":"jmp","labels":["mid"]}  ->  cond: true, x: 11, z: 20
{"op":"const","value":21,"dest":"z","type":"int"}  ->  cond: true, x: 10, z: 21
{"op":"const","value":11,"dest":"x","type":"int"}  ->  cond: true, x: 11, z: 21
{"op":"br","args":["cond"],"labels":["mid","br4"]}  ->  cond: true, x: 11, z: 21
{"op":"const","value":22,"dest":"z","type":"int"}  ->  cond: true, x: 11, z: 22
{"op":"br","args":["cond"],"labels":["br3","br4"]}  ->  cond: true, x: 11, z: 22
{"op":"id","args":["z"],"dest":"y","type":"int"}  ->  x: 12
{"op":"const","value":12,"dest":"x","type":"int"}  ->  x: 12
{"op":"jmp","labels":["end"]}  ->  x: 12
{"op":"id","args":["y"],"dest":"z","type":"int"}  ->  cond: true, x: ?, z: 21
{"op":"jmp","labels":["end"]}  ->  cond: true, x: ?, z: 21
{"op":"print","args":["x","y","z"]}  ->  cond: true, x: ?, z: 21
